src/engine/TextIndexScanForEntity.h

```diff
@@ -86,6 +105,16 @@ class TextIndexScanForEntity : public Operation {
   VariableToColumnMap computeVariableToColumnMap() const override;
 
  private:
+  const VocabIndex& getVocabIndexOfFixedEntity() const {
+    if (!hasFixedEntity()) {
+      AD_CONTRACT_CHECK(
+          "getVocabIndexOfFixedEntity called on an instance that does not have "
+          "a fixed entity and therefore also no VocabIndex.");
+    }
+    return std::get<std::pair<std::string, VocabIndex>>(varOrFixed_.entity_)
+        .second;
+  }
```

Comment:
All of those getters (the has and is and get functions) can be implemented in the `.cpp` file.

Reply:
(except for if they fit into two lines after refactorings.